pysam/libchtslib.pyx

```diff
@@ -222,6 +226,8 @@ cdef class HTSFile(object):
                 fd = self.filename
             else:
                 fd = self.filename.fileno()
+                
+            dup_fd = dup(fd)
```

Comment:
... on second thoughts, I wonder if it might get a bit confusing for the average user as they wonder if they should or should not call AlignmentFile.close(), etc.
Maybe being explicit is better such as "own_filehandle" or "duplicate_filehandle" for advanced user is clearer?